### DISCUSSION TITLE
Examples for how to read Map and Union Types from the C API

### DISCUSSION THREAD

**Discussion by kevinseekell** [duckdb/duckdb#13579] [Q&A] [answered]
I found the documentation page (https://duckdb.org/docs/api/c/vector) really helpful in learning how to handle Lists, Arrays and Structs from the DuckDB C API. The two nested types that appear to be missing from the page are Map and Union. 

I do see that the logical type interface provides functions to get the underlying key/value/member types, However, I don't see any functions for extracting the child vector from the parent vector. I suppose I was expecting to see a function such as:

duckdb_vector duckdb_map_vector_get_child_key(duckdb_vector vector);

Along with functions to get the meta data needed to interpret the data (ex. how many key/values does each map have?)

I could certainly be missing something though. If anyone could point me to some examples of how these two types should be read in I'd appreciate it.

**Comment by Maxxen** [comment, marked as answer]
Maps are just lists of {key: V, value: K} structs, and unions are just structs with a special "tag" field as first member, so you should be able tl jse the same functions as when accessing lists and structs.

**Reply by kevinseekell** [reply]
Thanks @Maxxen. I'll try this out.

**Reply by Giorgi** [reply]
If it helps, here is C# implementation for reading Map: https://github.com/Giorgi/DuckDB.NET/blob/develop/DuckDB.NET.Data%2FInternal%2FReader%2FMapVectorDataReader.cs
